src/main/webapp/js/components/interface/drawer/TabbedDrawer.js

```diff
@@ -79,90 +78,117 @@ define(function (require) {
             }, this);
             return renderedLabels;
-        }
+        };
 
         // function to render all the childs, wrap each one of them in a div and manage with display
         // which child has to be visible.
-        renderMyChildren() {
-            var renderedComponents = this.props.children.map(function(child, index) {
-                if(this.state.drawerOpened == true) {
-                    var ElementToRender = child;
-                    var MyElement = (this.state.selectedTab != index) ? <div className="hiddenComponent" key={index}><ElementToRender /></div> : <div key={index}><ElementToRender /></div>;
-
-                } else {
-                    var ElementToRender = child;
-                    var MyElement = <div key={index}><ElementToRender className="hiddenComponent" /></div>;
-                }
-                return (MyElement);
-            }, this);
-            return renderedComponents;
-        }
+        renderMyChildren = () => {
+            var paddingChildren = 45;
+            if (this.props.children != undefined) {
+                var renderedComponents = this.props.children.map(function (child, index) {
+                    var ElementToRender = undefined;
+                    var MyElement = undefined;
+                    if(child.type != undefined) {
+                        var ElementToRender = child.type;
+                        if (this.state.drawerOpened == true) {
+                            MyElement = (this.state.selectedTab != index) ? <div className="hiddenComponent" key={index}><ElementToRender iframeHeight={this.state.drawerHeight - paddingChildren} {...child.props} /></div> : <div key={index}><ElementToRender iframeHeight={this.state.drawerHeight - paddingChildren} {...child.props} /></div>;
+                        } else {
+                            MyElement = <div key={index}><ElementToRender className="hiddenComponent" iframeHeight={this.state.drawerHeight - paddingChildren} {...child.props} /></div>;
+                        }
+                    } else {
+                        var ElementToRender = child;
+                        if (this.state.drawerOpened == true) {
+                            MyElement = (this.state.selectedTab != index) ? <div className="hiddenComponent" key={index}><ElementToRender iframeHeight={this.state.drawerHeight - paddingChildren} {...this.props.childrenProps[index]} /></div> : <div key={index}><ElementToRender iframeHeight={this.state.drawerHeight - paddingChildren} {...this.props.childrenProps[index]} /></div>;
+                        } else {
+                            MyElement = <div key={index}><ElementToRender className="hiddenComponent" iframeHeight={this.state.drawerHeight - paddingChildren} {...this.props.childrenProps[index]} /></div>;
+                        }
+                    }
+                    return (MyElement);
+                }, this);
+                return renderedComponents;
+            } else {
```

Comment:
is this actually needed? if we remove this `else` won't we have the same behaviour?

Reply:
React itself triggers an error when children are not defined, since it tries to loop with map through the undefined (I guess), that is needed from my point of view.

Reply:
Yep, you are right!